### PR TITLE
fix trainer during saving checkpoints

### LINKED ISSUE #6575: [Bug]: 单卡微调gpt-3模型，出现GPTTrainer获取不到dp_group
### 软件环境

```Markdown
- paddlepaddle:2.5.0
- paddlepaddle-xpu: 2.5.0
- paddlenlp: 2.6.0rc0.post0
```

### 重复问题

- [X] I have searched the existing issues

### 错误描述

```Markdown
单卡微调时，通过paddle.distributed.get_world_size()获取world_size的值为1，
无法对dp_group属性进行赋值。只有当world_size大于1时，才会赋值。
导致保存模型参数时，会找不到属性报错。
[图片]
```

### 稳定复现步骤 & 代码

export PYTHONPATH="../../PaddleNLP/"

log_dir="log"
rm -rf $log_dir

python3 -m paddle.distributed.launch \
    --xpus "0" \
    --log_dir ${log_dir} \
    finetune_generation.py \
    --model_type "gpt" \
    --model_name_or_path gpt2-medium-en \
    --output_dir "output/$task_name" \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 1 \
    --tensor_parallel_degree 1 \
    --pipeline_parallel_degree 1 \
    --scale_loss 1024 \
    --learning_rate 0.00001 \
    --max_steps 10000 \
    --save_steps 5000 \
    --weight_decay 0.01 \
    --warmup_ratio 0.01 \
    --max_grad_norm 1.0 \
    --logging_steps 1\
    --dataloader_num_workers 1 \
    --sharding "stage2" \
    --max_evaluate_steps 1000 \
    --eval_steps 1000 \
    --report_to "visualdl" \
    --disable_tqdm true \
    --recompute 1 \
    --gradient_accumulation_steps 2 \
    --do_train \
    --do_eval \
    --device "xpu"
执行以上代码，问题必现。
解决方案：可以在获取属性之前，通过hasattr判断属性是否存在。

### PR DESCRIPTION
**Pull request by DrownFish19**
### PR types

Bug fixes

### PR changes

Others

### Description

* fix trainer during saving checkpoints for single gpu.
* This resolves #6575

### CODE REVIEW

File: paddlenlp/trainer/trainer.py [hunk 1]
```diff
@@ -1745,13 +1745,14 @@ def _save_checkpoint(self, model, metrics=None):
...
-                    os.path.join(output_dir, optimizer_name),
-                )
+        if paddle.distributed.get_world_size() > 1:
+            if self.args.use_hybrid_parallel:
+                if self.dp_group.rank <= 0:
+                    os.makedirs(output_dir, exist_ok=True)
+                    paddle.save(
+                        self.optimizer.state_dict(),
+                        os.path.join(output_dir, optimizer_name),
+                    )
```
Comment (DrownFish19) [inline paddlenlp/trainer/trainer.py]: 参考代码https://github.com/PaddlePaddle/PaddleNLP/blob/bc8df6ef875dab07862282c9d0ad22996c71a9e9/paddlenlp/trainer/trainer.py#L257-L261

File: paddlenlp/trainer/training_args.py [hunk 2]
```diff
@@ -766,8 +766,9 @@ def __post_init__(self):
         if len(self.sharding) == 0 and self.sharding_parallel_degree > 0:
             warnings.warn("`--sharding_parallel_degree` is useful only when `--sharding` is specified.")
 
-        if len(self.sharding) > 0 or self.tensor_parallel_degree > 1 or self.pipeline_parallel_degree > 1:
-            self.use_hybrid_parallel = True
+        if paddle.distributed.get_world_size() > 1:
+            if len(self.sharding) > 0 or self.tensor_parallel_degree > 1 or self.pipeline_parallel_degree > 1:
```
Comment (ZHUI) [inline paddlenlp/trainer/training_args.py]:
```suggestion
            if len(self.sharding) > 0 or self.tensor_parallel_degree > 1 or self.pipeline_parallel_degree > 1 and paddle.distributed.get_world_size() > 1:
```

File: paddlenlp/trainer/training_args.py [hunk 3]
```diff
@@ -766,7 +766,12 @@ def __post_init__(self):
         if len(self.sharding) == 0 and self.sharding_parallel_degree > 0:
             warnings.warn("`--sharding_parallel_degree` is useful only when `--sharding` is specified.")
 
-        if len(self.sharding) > 0 or self.tensor_parallel_degree > 1 or self.pipeline_parallel_degree > 1:
+        if (
+            len(self.sharding) > 0
+            or self.tensor_parallel_degree > 1
+            or self.pipeline_parallel_degree > 1
+            and paddle.distributed.get_world_size() > 1
+        ):
```
Comment (ZHUI) [inline paddlenlp/trainer/training_args.py]:
```suggestion
        if paddle.distributed.get_world_size() > 1 and (
            len(self.sharding) > 0
            or self.tensor_parallel_degree > 1
            or self.pipeline_parallel_degree > 1
        ):
```

**Review by ZHUI — APPROVED** [review]
LGTM

**Pull request merged**